engines/gob/inter_v7.cpp

```diff
@@ -1420,14 +1927,68 @@ void Inter_v7::o7_writeData(OpFuncParams &params) {
 		warning("Attempted to write to file \"%s\"", file.c_str());
 }
 
+Common::String Inter_v7::ansiToOEM(Common::String string) {
+	Common::U32String u32String = string.decode(Common::kWindows1252);
+	// Replace characters that do not exist in the target codepage with the closest match
+	for (int i = 0; i < (int) u32String.size(); ++i) {
+		// Replace curly double quotes with straight double quotes
+		if (u32String[i] == 0x201C || u32String[i] == 0x201D || u32String[i] == 0x201E) {
+			u32String.setChar(0x22, i);
+		}
+
+		// Replace curly single quotes with straight single quotes
+		if (u32String[i] == 0x2018 || u32String[i] == 0x2019) {
+			u32String.setChar(0x27, i);
+		}
+	}
+
+	return u32String.encode(Common::kDos850);
+}
+
+
+Common::String Inter_v7::oemToANSI(Common::String string) {
+	return string.decode(Common::kDos850).encode(Common::kWindows1252);
+}
+
+void Inter_v7::o7_ansiToOEM(OpGobParams &params) {
+	uint16 varIndex = _vm->_game->_script->readUint16();
+	char *str = GET_VAR_STR(varIndex);
+	Common::String oemString = ansiToOEM(Common::String(str));
+	WRITE_VAR_STR(varIndex, oemString.c_str());
+}
+
 
 void Inter_v7::o7_oemToANSI(OpGobParams &params) {
-	_vm->_game->_script->skip(2);
+	uint16 varIndex = _vm->_game->_script->readUint16();
+	char *str = GET_VAR_STR(varIndex);
+	Common::String ansiString = oemToANSI(Common::String(str));
+	WRITE_VAR_STR(varIndex, ansiString.c_str());
+}
+
+void Inter_v7::o7_setDBStringEncoding(OpGobParams &params) {
+	uint16 varIndex = _vm->_game->_script->readUint16();
+	int mode = READ_VAR_UINT32(varIndex);
+	switch (mode) {
+	case 0:
+	case 1:
+		_translationDatabases.setEncodingIsOEM(mode);
+		break;
+	default:
+		warning("o7_setDBStringEncoding: invalid mode %d", mode);
+		break;
+	}
 }
 
 void Inter_v7::o7_gob0x201(OpGobParams &params) {
 	uint16 varIndex = _vm->_game->_script->readUint16();
 
 	WRITE_VAR(varIndex, 1);
 }
+
+void Inter_v7::o7_getFreeDiskSpace(OpGobParams &params) {
+	uint16 varIndex = _vm->_game->_script->readUint16();
+	WRITE_VAR(varIndex, 1000000000); // HACK
```

Comment:
Out of interest, what is this used for?

Reply:
It's an engine opcode originally used by the game scripts to check the amount of available hard disk space. When starting Adibou2/Sciences the very first time, the script performed an "installation" step, where it copied some data from the CD to the hard disk, and called this opcode beforehand to check if there was enough disk space for that. If not, an error message was displayed and the game was stopped.

Here in ScummVM we just announce a fake fixed amount of free disk space, high enough to make the game scripts happy (and this is safe, as there is no actual copy taking place here from any "CD", all the data is already in the game directory).

Reply:
Thanks for the explanation. Could you add a comment with the explanation for reference?

Reply:
Added in master.